Issue/1600 doc only for registry API

t83714 opened the pull request:
### What this PR does

Fixes #1600 

### Checklist

-   [x] unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)
-   [x] I've assigned this PR to someone (if you don't know who to assign it to, pick @AlexGilleran)

AlexGilleran commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala:
```diff
@@ -100,6 +141,20 @@ class RecordsService(config: Config, webHookActor: ActorRef, authClient: AuthApi
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    *
+    * @apiSuccess (Success 200) {json} Response the record created
+    * @apiError (Error 400) {json} could not create
```
This turns into 

![image](https://user-images.githubusercontent.com/900555/44707771-cc359b80-aae8-11e8-80fe-61c3d6ecff93.png)

t83714 replied:
Updated~

AlexGilleran commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala:
```diff
@@ -280,6 +412,18 @@ class RecordsService(config: Config, webHookActor: ActorRef, authClient: AuthApi
+    * @apiDescription The patch should follow IETF RFC 6902 (https://tools.ietf.org/html/rfc6902).
+    *
+    * @apiParam (path) {string} id ID of the aspect to be saved.
+    * @apiParam (body) {string} recordPatch The RFC 6902 patch to apply to the aspect.
```
I don't think this is a string 😬

t83714 replied:
Updated~

AlexGilleran commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala:
```diff
@@ -166,6 +245,21 @@ class RecordsService(config: Config, webHookActor: ActorRef, authClient: AuthApi
+    *
+    * @apiSuccess (Success 200) {json} Response the trim result
+    * @apiSuccess (Success 202) {string} Response Deletion is taking a long time (normal for sources with many records) but it has worked
+    * @apiError (Error 400) {string} The records could not be deleted, possibly because they are used by other records.
```
Turns into

![image](https://user-images.githubusercontent.com/900555/44707911-246c9d80-aae9-11e8-96b2-1fa7984987c1.png)

t83714 replied:
Updated~

AlexGilleran commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -120,6 +164,20 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    * @apiParam (path) {string} id ID of the web hook to delete.
+    *
+    * @apiSuccess (Success 200) {json} Response deletion result
+    * @apiError (Error 400) {json} could not delete
```
Turns to could | not delete

t83714 replied:
Updated~

t83714 commented:
@AlexGilleran 
Updated the documents~
I think it's ready to have a look agin~
Thank you~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala:
```diff
@@ -19,6 +19,13 @@ import com.typesafe.config.Config
+/**
+  * @apiGroup Registry Aspects
+  * @api {get} /v0/registry/aspects Get a list of all aspects
+  * @apiSuccess (Success 200) {json} Response The aspect definitions.
```
@AlexGilleran 
Sorry~ forget about the example thing~ my bad~
Will added in~

t83714 replied:
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala:
```diff
@@ -55,6 +74,18 @@ class AspectsService(config: Config, authClient: AuthApiClient, webHookActor: Ac
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the aspect to be fetched.
+    *
+    * @apiSuccess (Success 200) {json} Response The details of the aspect.
```
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala:
```diff
@@ -72,6 +103,19 @@ class AspectsService(config: Config, authClient: AuthApiClient, webHookActor: Ac
+    * @apiParam (path) {string} id ID of the aspect to be saved.
+    * @apiParam (body) {json} aspect The aspect to save.
+    *
+    * @apiSuccess (Success 200) {json} Response The details of the aspect saved.
```
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -21,7 +21,15 @@ import scala.concurrent.duration._
+  *
+  * @apiDescription Get a list of all web hooks
+  * @apiHeader {string} X-Magda-Session Magda internal session id
+  * @apiSuccess (Success 200) {json} Response a list of webhook records
```
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -48,6 +56,19 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (body) {Object} webhook An object contains all webhook information
+    *
+    * @apiSuccess (Success 200) {json} Response the created webhook record
```
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -68,6 +89,16 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    * @apiDescription Get a web hook by ID
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to be fetched.
+    * @apiSuccess (Success 200) {json} Response the webhook record
```
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala:
```diff
@@ -99,6 +143,19 @@ class AspectsService(config: Config, authClient: AuthApiClient, webHookActor: Ac
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the aspect to be saved.
+    * @apiParam (body) {json} aspectPatch The RFC 6902 patch to apply to the aspect.
```
Updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -48,6 +56,19 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    * @apiDescription Create a new web hook
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (body) {Object} webhook An object contains all webhook information
```
updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -95,6 +126,19 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to be fetched.
+    * @apiParam (body) {Object} webhook The web hook to save.
```
updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -120,6 +164,20 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to delete.
+    *
+    * @apiSuccess (Success 200) {json} Response deletion result
```
updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala:
```diff
@@ -143,6 +201,18 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to be acknowledged.
+    *
+    * @apiSuccess (Success 200) {json} Response The details of the acknowledgement.
```
updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordAspectsService.scala:
```diff
@@ -19,6 +19,16 @@ import scala.util.Success
+  *
+  * @apiDescription Get a list of all aspects of a record
+  * @apiParam (path) {string} recordId ID of the record for which to fetch aspects.
+  * @apiSuccess (Success 200) {json} Response a list of aspects
```
updated~

t83714 commented on magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordAspectsService.scala:
```diff
@@ -58,6 +78,19 @@ class RecordAspectsService(webHookActor: ActorRef, authClient: AuthApiClient, sy
+    * @apiParam (body) {json} aspect The record aspect to save
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    *
+    * @apiSuccess (Success 200) {json} Response the aspect detail
```
updated~

t83714 commented:
@AlexGilleran is ready to have a look again~
I think what you want probably is to make sure all `JSON` type `response` & `param` all have an example section --- I have updated all endpoints to make sure that 😄 
There was one ErrorReponse used to be `JSON` -- and it was a mistake
I've changed it to `string` --- we don't have a structured response for that one~
Understand this doc may be useful for coming `GovHack` event~ 
Happy to update further if we can do it better~ 😄
